python/paddle/v2/fluid/backward.py

```diff
@@ -238,8 +299,17 @@ def empty_callback(block, context):
         if op.has_attr("sub_block"):
             sub_block = program.block(op.block_attr("sub_block"))
             grad_sub_block = program.create_block(parent_idx=sub_block.idx)
-            _append_backward_ops_(sub_block, sub_block.ops, grad_sub_block,
-                                  no_grad_dict, grad_to_var)
+            cb = _callback_lookup_(op)
+            if cb is not None:
+                if callbacks is None:
+                    new_callbacks = [cb]
+                else:
+                    new_callbacks = callbacks + [_callback_lookup_(op)]
+                _append_backward_ops_(sub_block, sub_block.ops, grad_sub_block,
+                                      no_grad_dict, grad_to_var, new_callbacks)
```

Comment:
Do we need to do `callbacks = new_callbacks`, since `callbacks` is used later as well.

Reply:
`callbacks` used later should not contain `_callback_lookup_(op)`